src/Gui/MainWindowPy.cpp

```diff
@@ -215,3 +224,57 @@ Py::Object MainWindowPy::removeWindow(const Py::Tuple& args)
     }
     return Py::None();
 }
+
+Py::Object MainWindowPy::showHint(const Py::Tuple& args)
+{
+    static auto userInputFromPyObject = [](const Py::Object& object) -> InputHint::UserInput {
+        Py::Long value(object.getAttr("value"));
+
+        return static_cast<InputHint::UserInput>(value.as_long());
```

Comment:
This won't crash or cause any runtime issue, though just noting that it can be out of bounds

Reply:
I am aware of it but I don't know of any way of safe casting the enum class from long. Do you have any suggestion on how to do that better? It would be nice to add such tool into my toolbelt.

Reply:
That solution is impossible because domain of values for this enum is not continous. I'll play with the idea of doing it more data driven as suggested in other comments and see how it will play out.

I'll mark this as resolved then as either it will be completely changed or we leave it as is.